scheduler/src/cook/kubernetes/api.clj

```diff
@@ -643,6 +643,24 @@
                                      (.withEmptyDir (V1EmptyDirVolumeSource.)))]
     (.build builder)))
 
+(defn make-shm-volume
+  "Make a volume for /dev/shm and add it to the volume mount for the given image"
+  []
+  (let [^V1EmptyDirVolumeSource volume-source (V1EmptyDirVolumeSource.)
+        _ (.medium volume-source "Memory")
```

Comment:
Do we need to account for this in the memory usage of the pod?

Reply:
AFAIK, no you don't specify a size here. I checked and it appears that the kernel/system limit is all RAM. (I also seem to recall from discussions that this gets accounted for in cgroups. which makes sense, this shm isn't shared among pods.)

Reply:
So does this use the same pool of memory as what you ask for on the container?

Reply:
https://www.kernel.org/doc/Documentation/cgroup-v1/memory.txt suggests yes as the creating cgroup container would own it and get it accounted.

Reply:
Would it be possible to write a test where:

1. job requests 1024 MiB with the shared mem label
2. command writes 768 MiB to `/dev/shm/foo`
3. command then tries to allocate 512 MiB of memory
4. assert that the job gets OOM

Reply:
/dev is almost assuregly a tmpfs (in modern linux with udev), so that OOM would occur whether it was shm or tmpfs, so I would expect this to fail with or without the label.

Reply:
Ah, got it.